streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1543,26 +1544,28 @@ private boolean close(final Optional<Long> timeout, final boolean leaveGroup) {
             timeoutMs = Long.MAX_VALUE;
         }
 
-        if (state.hasCompletedShutdown()) {
```

Comment:
Seems we could still keep this condition? Even if I agree, we might not need to handler this case upfront any longer.